### PR TITLE
Use correct casing for cmdlet name and cmdlet parameter name in *.ps1 files

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

* Apply PSScriptAnalyzer `PSUseCorrectCasing` rule.
* Reverted chnages to `demos\DSC\dsc-demo.ps1`, see https://github.com/PowerShell/PSScriptAnalyzer/issues/1474

## PR Context

PSScriptAnalyzer v1.19.0 introduces Cmdlet parameter name casing.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Item.Tests.ps1 [hunk 1]
```diff
@@ -31,9 +31,9 @@ Describe "Rename-Item tests" -Tag "CI" {
         $oldSp = "$wdSp/$oldSpBName"
         $newSpName = "[renamed]file.txt"
         $newSp = "$wdSp/``[renamed``]file.txt"
-        In $wdSp -Execute {
+        In $wdSp -execute {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Item.Tests.ps1:34]:
```suggestion
        In $wdSp -Execute {
```

File: test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Item.Tests.ps1 [hunk 2]
```diff
@@ -31,9 +31,9 @@ Describe "Rename-Item tests" -Tag "CI" {
...
             $null = New-Item -Name $oldSpName -ItemType File -Value $content -Force
-            Rename-Item -Path $oldSpBName $newSpName
+            Rename-Item -path $oldSpBName $newSpName
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Item.Tests.ps1:36]:
```suggestion
            Rename-Item -Path $oldSpBName $newSpName
```

File: test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Item.Tests.ps1 [hunk 3]
```diff
@@ -46,7 +46,7 @@ Describe "Rename-Item tests" -Tag "CI" {
         $oldSp = "$wdSp/$oldSpBName"
         $newSpName = "[renamed]file2.txt"
         $newSp = "$wdSp/``[renamed``]file2.txt"
-        In $wdSp -Execute {
+        In $wdSp -execute {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Item.Tests.ps1:49]:
```suggestion
        In $wdSp -Execute {
```

File: test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage2.Tests.ps1 [hunk 4]
```diff
@@ -1,4 +1,4 @@
-﻿# Copyright (c) Microsoft Corporation.
+# Copyright (c) Microsoft Corporation.
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage2.Tests.ps1:1]: What is changed here?

**Review by iSazonov — COMMENTED** [review]
Compare operators should be in low case.

File: build.psm1 [hunk 5]
```diff
@@ -1469,7 +1469,7 @@ function Test-XUnitTestResults
         throw "Cannot convert $TestResultsFile to xml : $($_.message)"
     }
 
-    $failedTests = $results.assemblies.assembly.collection | Where-Object failed -gt 0
+    $failedTests = $results.assemblies.assembly.collection | Where-Object failed -GT 0
```
[the remark this answers is not in the document]
Comment (xtqqczze) [reply build.psm1:1472]: This is the correct capitalization for these parameters in `Where-Object`. I didn't expect this either!

File: test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage2.Tests.ps1 (hunk 4 above)
Comment (xtqqczze) [reply test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage2.Tests.ps1:1]: Removed UTF BOM.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Item.Tests.ps1 (hunk 2 above)
Comment (xtqqczze) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Item.Tests.ps1]: @bergmeister this is unexpected

File: test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Item.Tests.ps1 (hunk 1 above)
Comment (xtqqczze) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Item.Tests.ps1:34]: This is correct, but opened upstream PR: https://github.com/pester/Pester/pull/1523

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
@xtqqczze In future please make one change per commit - this greatly speeds up a code review.

Also it would be nice to have static tests for this to exclude regressions in formatting.

**Review by iSazonov — APPROVED** [review]

**Pull request merged**